### ISSUE TITLE
bug: false positive no-pass-data-to-parent when useReducer initial state references a hook parameter

### ISSUE THREAD

**Issue by victorpi463** [nickjvandyke/eslint-plugin-react-you-might-not-need-an-effect#63]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.9.1

### ESLint version

9.28.0

### Describe the bug

`no-pass-data-to-parent` flags `dispatch()` inside a `useEffect` when the `useReducer` initial state references a custom hook parameter, even though `dispatch` is local and the dispatched action is a constant.

The rule traces `dispatch` -> `useReducer(reducer, initialState)` -> finds a hook parameter (`initialData`) inside `initialState` -> concludes that the effect is "passing data to a parent." But the dispatched action `{ type: "FETCH" }` doesn't contain any prop data. `dispatch` updates local state within the hook,  it does not pass anything to a parent.

Removing `initialData` from the initial state makes the warning disappear, confirming that the trigger is the presence of a hook parameter in `useReducer`'s second argument, not the actual effect behavior.

### Steps To Reproduce

  ```tsx
  import { useEffect, useReducer } from "react"

  type State = { data: string | undefined }
  type Action = { type: "FETCH" }

  function reducer(state: State, action: Action): State {
    switch (action.type) {
      case "FETCH":
        return { ...state, data: "loading" }
    }
  }

  export function useData({ initialData, enabled = true }: { initialData?: string; enabled?: boolean }) {
    const [state, dispatch] = useReducer(reducer, { data: initialData })

    useEffect(() => {
      if (!enabled) return
      dispatch({ type: "FETCH" }) // ⚠️ no-pass-data-to-parent
    }, [enabled])

    return state
  }
```

  - dispatch`({ type: "FETCH" })` is flagged
  - Replace `{ data: initialData }` with `{ data: undefined }` -> warning disappears
  - Change back to `{ data: initialData }` -> warning reappears

### Expected Behavior

No warning. dispatch is local to the hook (from useReducer), and the dispatched action is a constant object `{ type: "FETCH" }`. No data is being passed to a parent component, the hook manages its own internal state.
